src/common/structures/aruco_board_specification.py

```diff
@@ -0,0 +1,48 @@
+from typing import Any, Dict, List
+from pydantic import BaseModel, validator
+from .vec3 import Vec3
+import json, hjson
+
+class BoardMarker(BaseModel):
+    marker_id: int
+    points: List[Vec3]
+
+    @validator('points', pre=True, each_item=True)
+    def validate_points(cls, value):
+        if isinstance(value, dict):
+            return Vec3(**value)
+        if isinstance(value, Vec3):
+            return value
+        raise ValueError("Invalid type for point")
```

Comment:
I think you should be ok without this explicit check. pydantic normally won't let you create a data structure with a member whose data type doesn't match what you've specified.

It might make sense to check that `points` is a list of length 4 (assuming we are using ArUco)? If so, then can you also add a tiny test to make sure this validation works as intended?